slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -192,21 +206,12 @@ std::string SLAMServiceImpl::PaintMap(bool pose_marker_flag) {
     }
 
     for (const auto &submap_id_pose : submap_poses) {
-        cartographer::mapping::proto::SubmapQuery::Response response_proto;
-
-        {
-            std::lock_guard<std::mutex> lk(map_builder_mutex);
-            const std::string error = map_builder.map_builder_->SubmapToProto(
-                submap_id_pose.id, &response_proto);
-            if (error != "") {
-                throw std::runtime_error(error);
-            }
-        }
-
         auto submap_textures =
             absl::make_unique<::cartographer::io::SubmapTextures>();
-        submap_textures->version = response_proto.submap_version();
-        for (const auto &texture_proto : response_proto.textures()) {
+        submap_textures->version =
+            response_protos[submap_id_pose.id].submap_version();
+        for (const auto &texture_proto :
```

Comment:
[nit] `const auto &&`

Reply:
for whatever reason this one was throwing an error, so changed it back to a single &